tests/utils.py

```diff
@@ -307,3 +307,17 @@ def popen_karapace_all(config_path: Union[Path, str], stdout: IO, stderr: IO, **
     kwargs["stdout"] = stdout
     kwargs["stderr"] = stderr
     return Popen([python_exe(), "-m", "karapace.karapace_all", str(config_path)], **kwargs)
+
+
+class StubMessage:
+    """A stub to stand-in for `confluent_kafka.Message` in unittests.
+
+    Since that class cannot be instantiated, thus this is a liberal simulation
+    of its behaviour ie. its attributes are accessible via getter functions:
+    `message.offset()`."""
+
+    def __init__(self, **attrs: Any) -> None:
+        self._attrs = attrs
+
+    def __getattr__(self, key: str) -> None:
+        return lambda: self._attrs[key]
```

Comment:
nit (non-blocking): I assume this is safe, but given code-base will evolve, would it be safer to re-raise as `AttributeError` on a failing access here?